src/common/src/config.rs

```diff
@@ -1131,7 +1127,7 @@ pub mod default {
 
     pub mod auto_dump_heap_profile {
         pub fn dir() -> String {
-            "".to_string()
+            ".".to_string() // current directory
```

Comment:
`"."` is not perfect but `./.risingwave` is only used by risedev (for developers), it seems even worse.

Let me try to make it configurable by env var, so that `kube-bench` or `risedev` can set a proper output directory.

Reply:
Also the ci failed because of this change. You need to update the example.toml

Reply:
> It's actually not just used by risedev.
> 
> https://github.com/risingwavelabs/risingwave/blob/840114a579ea18414324553abb6a18ef69e6aa5f/src/storage/src/memory.rs#L106
> 
> But an env for the prefix will be great. It can be useful for putting all kinds of local files.

Hmm, this is not persuasive. `memory` storage is just for test and playground is never used in production.

Reply:
At least put it in a directory? Output heap profile in a root directory looks ugly to me. Also we will have a directory for manually dumped in the dashboard pr. 

Anyway, I think an env var for the local file prefix would be great. Maybe we can do it in later prs.

Reply:
I just realized the "env var" actually already exist - `MALLOC_CONF`

Now it will dump memory profile to `prof.prefix` if `server.auto_dump_heap_profile.dir` is absent. Please take a look.  🥰